### DISCUSSION TITLE
Open souce english dictionary

### DISCUSSION THREAD

**Discussion by c0d3kraft** [TheOpenDictionary/odict#739] [Q&A]
Are there any open source english dictionaries available in the odict format, that I can use in an application? Also wanted to know if odict has support for bi-lingual dictionaries.

**Comment by Nickersoft** [comment]
Hey @c0d3kraft! The ODict organization has a repo of open-source dictionary converters that will download various open-source dictionaries and convert them to the ODict format. There is an English Wiktionary format supported there.

These files aren't hosted anywhere yet though, so you'll have to download Python and run the scripts themselves: https://github.com/TheOpenDictionary/dictionaries.

Also – you can support bilingual dictionaries by just having your Entry's `term` field be in one language and the definitions be in another :)

**Reply by c0d3kraft** [reply]
Thanks @Nickersoft , I was able to get the English wiktionary, following the instructions in the dictionaries repo. And yes bi-lingual is alreadty there, as you explained. Is there any support for declensions in odict or do I need a create a separate term for each form of the word. I am working on some Indic language dictionaries - where each word can have many forms (often 4-5 on average), keeping each declension as a term would make the dict files quite big.

**Reply by Nickersoft** [reply]
Neat to hear that you're doing something cool! Glad you got it sorted :)

Declensions aren't currently supported, but we could always extend the format to support them. That's why it's open-source, so it can be driven by the community needs :)

What would your requirements be? To just list the declensions under the parent term? Or does each declension need to have a note or definition with it?

**Reply by c0d3kraft** [reply]
The primary requirement is that I should be able to do a lookup for a declension of a word. If the term in dictionary is _cook_ and I do a lookup for _cooking_ or _cooked_ , the system should return the definition associated with _cook_ , provided that _cooking_ and _cooked_ are defined as declensions of the term _cook_

**Reply by Nickersoft** [reply]
Oh! This is actually already possible by using the "see" attribute along with the "follow" option. You can alias entry stubs to other entries and tell ODict to just return the original definition.

https://www.odict.org/docs/xml/entry
https://www.odict.org/docs/cli/lookup

**Comment by tahir-hassan** [comment]
@Nickersoft Thanks for the great work.  I want to use odict as an offline CLI dictionary that I can script in various ways. 

I have gone to https://github.com/TheOpenDictionary/dictionaries but there is no `README` therein that describes how to get an English dictionary up and running.  

Can you provide at least an English dictionary to make the process easier for new users?  If you can't for copyright reasons, can you please provide clear instructions?

**Reply by Nickersoft** [reply]
Hey Tahir! Just updated the docs on the repo. Hope that clears things up for you!

I'm actually working on ODict V3 right now, which will allow you to download pre-generated dictionaries from my remote R2 bucket. So in JavaScript, for example:

```javascript
const dictionary = await OpenDictionary.load("wiktionary/eng");
dictionary.lookup("hello");
```
